KeyboardSupport/Classes/KeyboardSupportViewController.swift

```diff
@@ -0,0 +1,188 @@
+//
+//  KeyboardSupportViewController.swift
+//  KeyboardSupport
+//
+//  Copyright © 2018 Bottle Rocket Studios. All rights reserved.
+//
+
+import UIKit
+
+/// Callback when the "Done" button is tapped.
+public protocol KeyboardSupportDelegate: class {
+    func didTapDoneButton()
+}
+
+/// A model composed of different options for supporting the keyboard.
+public struct KeyboardSupportConfiguration {
+    public var textFields: [UITextField] = []
+    public var scrollView: UIScrollView?
+    public var bottomConstraint: NSLayoutConstraint?
+    public var constraintOffset: CGFloat = 0
+    public var usesDismissalView: Bool = false
+    public var usesKeyboardNextButtons: Bool = false
+    public var keyboardInputAccessoryView: KeyboardInputAccessoryView?
+    
+    public init(textFields: [UITextField] = [], scrollView: UIScrollView? = nil, bottomConstraint: NSLayoutConstraint? = nil, constraintOffset: CGFloat = 0, usesDismissalView: Bool = false, usesKeyboardNextButtons: Bool = false, keyboardInputAccessoryView: KeyboardInputAccessoryView? = nil) {
+        self.textFields = textFields
+        self.scrollView = scrollView
+        self.bottomConstraint = bottomConstraint
+        self.constraintOffset = constraintOffset
+        self.usesDismissalView = usesDismissalView
+        self.usesKeyboardNextButtons = usesKeyboardNextButtons
+        self.keyboardInputAccessoryView = keyboardInputAccessoryView
+    }
+}
+
+/// A subclass of UIViewController for making keyboard interactions easier.
+open class KeyboardSupportViewController: UIViewController {
```

Comment:
Let's keep a close eye on how this works out. I feel a bit uneasy about forcing clients to inherit from a special type of view controller, but we may find that it's rarely an issue in production.